src/tests/functional/shared_test_classes/src/single_op/multiclass_nms.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/single_op/multiclass_nms.hpp"
+
+#include "common_test_utils/test_enums.hpp"
+#include "common_test_utils/ov_tensor_utils.hpp"
+
+namespace ov {
+namespace test {
+std::string MulticlassNmsLayerTest::getTestCaseName(const testing::TestParamInfo<MulticlassNmsParams>& obj) {
+    std::vector<InputShape> shapes;
+    InputTypes input_types;
+    int32_t nmsTopK, backgroundClass, keepTopK;
+    element::Type outType;
+    op::util::MulticlassNmsBase::SortResultType sortResultType;
+    InputfloatVar inFloatVar;
+    InputboolVar inboolVar;
+    std::string targetDevice;
+
+    std::tie(shapes, input_types, nmsTopK, inFloatVar, backgroundClass, keepTopK, outType, sortResultType, inboolVar, targetDevice) = obj.param;
+
+    ElementType paramsPrec, roisnumPrec;
+    std::tie(paramsPrec, roisnumPrec) = input_types;
+
+    float iouThr, scoreThr, nmsEta;
+    std::tie(iouThr, scoreThr, nmsEta) = inFloatVar;
+
+    bool sortResCB, normalized;
+    std::tie(sortResCB, normalized) = inboolVar;
+
+    std::ostringstream result;
+    result << "IS=(";
+    for (const auto& shape : shapes) {
+        result << ov::test::utils::partialShape2str({shape.first}) << "_";
+    }
+    result << ")_TS=(";
+    for (const auto& shape : shapes) {
+        for (const auto& item : shape.second) {
+            result << ov::test::utils::vec2str(item) << "_";
+        }
+    }
+
+    using ov::test::utils::operator<<;
+    result << ")_paramsPrec=" << paramsPrec << "_roisnumPrec=" << roisnumPrec << "_";
+    result << "nmsTopK=" << nmsTopK << "_";
+    result << "iouThr=" << iouThr << "_scoreThr=" << scoreThr << "_backgroundClass=" << backgroundClass << "_";
+    result << "keepTopK=" << keepTopK << "_outType=" << outType << "_";
+    result << "sortResultType=" << sortResultType << "_sortResCrossBatch=" << sortResCB << "_nmsEta=" << nmsEta << "_normalized=" << normalized << "_";
+    result << "TargetDevice=" << targetDevice;
+    return result.str();
+}
+
+void MulticlassNmsLayerTest::generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) {
+    inputs.clear();
+
+    const auto& funcInputs = function->inputs();
```

Comment:
should be migrated to `generate_inputs` in the near future. Won't block the PR according OV1.0 removing priority. Anyway, we can reuse static shapes in generate_inputs